docs/source/guides/quickstart.md

```diff
@@ -1,48 +1,195 @@
+Hardware wallet addresses will also be listed here if one is attached to the system hardware.
+Note that the Geth console does not return EIP-55 compliant checksum addresses, and instead will output
+the *lowercase* version of the address.  Since Nucypher requires EIP-55 checksum addresses, you will need 
+to convert your address to checksum format:
 
-If you're connecting to the `devnet`, you should use `--network devnet --teacher 18.222.119.242:9151`.
+```bash
+> web3.toChecksumAddress(eth.accounts[0])
+"0x287A817426DD1AE78ea23e9918e2273b6733a43D"
+```
 
-### Run a Geth-Connected Development Ursula
+## Connecting to The NuCypher Network
 
-Run a local geth node in development mode:
+### Connecting Nucypher to an Ethereum Provider
 
-```bash
-$ geth --dev
+```python
+from nucypher.blockchain.eth.interfaces import BlockchainInterfaceFactory
+BlockchainInterfaceFactory.initialize_interface(provider_uri='~/.ethereum/goerli/geth.ipc')
 ```
 
-Run a local development Ursula connected to the geth node
+### Ursula: Untrusted Re-Encryption Proxies
 
-```bash
-$ nucypher ursula run --dev --provider ipc:///tmp/geth.ipc --checksum-address <GETH_DEV_ADDRESS>
+When initializing an `Alice`, `Bob`, or `Ursula`, an initial "Stranger-`Ursula`" is needed to perform 
+the role of a `Teacher`, or "seednode":
+
+```python
+from nucypher.characters.lawful import Ursula, Alice
+
+ursula = Ursula.from_seed_and_stake_info(seed_uri=<URI>, federated_only=False)  # ie. https://0.0.0.0:9151
+another_ursula = Ursula.from_seed_and_stake_info(seed_uri=<URI>, federated_only=False)
+```
+
+Stranger `Ursula`s can be created by invoking the `from_seed_and_stake_info` method, then a `list` of `known_nodes`
+can be passed into any `Character`'s init. The `known_nodes` will inform your character of all of the nodes
+they know about network-wide, then kick-off the automated node-discovery loop:
+
+```python
+alice = Alice(known_nodes=[ursula, another_ursula], ...)
+```
+
+## Alice: Grant Access to a Secret
+
+```python
+from nucypher.characters.lawful import Alice, Bob, Ursula
+from nucypher.network.middleware import RestMiddleware
+from nucypher.blockchain.eth.registry import InMemoryContractRegistry
+
+# Application Side-Channel
+# --------------------------
+# bob_encrypting_key = <Side Channel>
+# bob_verifying_key = <Side Channel>
+
+ursula = Ursula.from_seed_and_stake_info(seed_uri='https://0.0.0.0:9151', federated_only=False)
+alice = Alice(known_nodes=[ursula],
+              checksum_address="0x287A817426DD1AE78ea23e9918e2273b6733a43D", 
+              registry=InMemoryContractRegistry.from_latest_publication(),
+              network_middleware=RestMiddleware())
+bob = Bob.from_public_keys(verifying_key=bob_verifying_key, encrypting_key=bob_encrypting_key)
+```
+
+```python
+from datetime import datetime
+import maya
+
+#
+# Grant
+#
+
+policy_end_datetime = maya.now() + datetime.timedelta(days=5)  # Five days from now
+policy = alice.grant(bob,
```

Comment:
```python
policy = alice.grant(bob,
                     label=b'my-secret-stuff',
                     m=2, n=3,
                     expiration=policy_end_datetime,
                     value=10 ** 16 // (5 * 3) * (5 * 3))
```
That's for `n=3`, 5 days and the total price being about `10 ** 16` wei

Reply:
Not sure if this is a suggestion or....